### PR TITLE
CORS-2991: post-installation check of userTags of compute machines and disks

### PR DESCRIPTION
**Pull request by jianli-wei**
- post-installation check of userTags of compute machines and disks
- remove "-tp-" keyword and featureSet settings for the 4.17 tests
- replace 3 tags of organization/54643501348 with project "openshift-qe" ones, because the service account stops having permissions to access these organization tags (see [test](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/48249/rehearse-48249-periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-gcp-ipi-labels-tags-filestore-csi-fips-amd-f28-destructive/1825357728038522880))
```
$ gcloud resource-manager tags keys describe 54643501348/ServicePhase
ERROR: (gcloud.resource-manager.tags.keys.describe) [ipi-xpn-minpt-permissions@openshift-qe.iam.gserviceaccount.com] does not have permission to access tagKeys instance [namespaced] (or it may not exist): The caller does not have permission. This command is authenticated as ipi-xpn-minpt-permissions@openshift-qe.iam.gserviceaccount.com which is the active account specified by the [core/account] property
$ gcloud resource-manager tags keys describe 54643501348/CostCenter
ERROR: (gcloud.resource-manager.tags.keys.describe) [ipi-xpn-minpt-permissions@openshift-qe.iam.gserviceaccount.com] does not have permission to access tagKeys instance [namespaced] (or it may not exist): The caller does not have permission. This command is authenticated as ipi-xpn-minpt-permissions@openshift-qe.iam.gserviceaccount.com which is the active account specified by the [core/account] property
$ gcloud resource-manager tags keys describe 54643501348/AppCode
ERROR: (gcloud.resource-manager.tags.keys.describe) [ipi-xpn-minpt-permissions@openshift-qe.iam.gserviceaccount.com] does not have permission to access tagKeys instance [namespaced] (or it may not exist): The caller does not have permission. This command is authenticated as ipi-xpn-minpt-permissions@openshift-qe.iam.gserviceaccount.com which is the active account specified by the [core/account] property
$

### CODE REVIEW

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-gcp-ipi-labels-tags-filestore-csi-fips-amd-f28-destructive

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-gcp-ipi-labels-tags-filestore-csi-fips-amd-f28-destructive

**Comment by jianli-wei** [comment]
Test name | Commit | Details | Comment
-- | -- | -- | --
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-gcp-ipi-labels-tags-filestore-csi-fips-amd-f28-destructive | https://github.com/openshift/release/commit/a910cef8bdcc86fc002aa6df922297b6a3bcf991 | [link](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/48249/rehearse-48249-periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-gcp-ipi-labels-tags-filestore-csi-fips-amd-f28-destructive/1825380308845858816) | "failed to initialize the cluster: Cluster operators monitoring, storage are not available", [OCPBUGS-38219](https://issues.redhat.com/browse/OCPBUGS-38219) (see [log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/48249/rehearse-48249-periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-gcp-ipi-labels-tags-filestore-csi-fips-amd-f28-destructive/1825380308845858816/artifacts/gcp-ipi-labels-tags-filestore-csi-fips-amd-f28-destructive/gather-extra/artifacts/pods/openshift-cluster-csi-drivers_gcp-pd-csi-driver-controller-775c66b75-m8297_csi-driver.log))

**Comment by jianli-wei** [comment]
@jianlinliu @gpei please help review, thanks!

File: ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh [hunk 1]
```diff
@@ -29,46 +29,83 @@ then
...
+  fi
+done
 
 echo "$(date -u --rfc-3339=seconds) - Checking userTags of image-registry buckets..."
 readarray -t items < <(gsutil ls | grep "${INFRA_ID}-image-registry")
 for line in "${items[@]}"; do
   name=$(basename "${line}")
   current_tags="$(gcloud resource-manager tags bindings list --parent=//storage.googleapis.com/projects/_/buckets/${name} --location=${GCP_REGION} --effective)"
   echo "${current_tags}"
-  validate_user_tags "${current_tags}" || validation_ret=$?
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh:64]: I wonder why we changed the checking way of `validate_user_tags` by return value?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__multi-nightly.yaml [hunk 2]
```diff
@@ -1861,22 +1861,21 @@ tests:
...
       OO_INDEX: quay.io/openshift-qe-optional-operators/aosqe-index:v4.17
     env:
-      FEATURE_SET: TechPreviewNoUpgrade
       FIPS_ENABLED: "true"
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-gcp-ipi-user-labels-tags-filestore-csi
-- as: gcp-ipi-user-labels-tags-filestore-csi-tp-arm-f14
-  cron: 13 9 10,26 * *
+- as: gcp-ipi-user-labels-tags-filestore-csi-arm-f14
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__multi-nightly.yaml:1877]:
```suggestion
- as: gcp-ipi-labels-tags-filestore-csi-arm-f14
```

To keep consistent name with the destructive job name.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__multi-nightly.yaml [hunk 3]
```diff
@@ -1861,22 +1861,21 @@ tests:
     test:
     - chain: openshift-e2e-test-qe
     workflow: cucushift-installer-rehearse-gcp-ipi-sno
-- as: gcp-ipi-labels-tags-filestore-csi-fips-tp-amd-f28-destructive
-  cron: 32 16 7 * *
+- as: gcp-ipi-labels-tags-filestore-csi-fips-amd-f28-destructive
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__multi-nightly.yaml:1864]: Can you also help update `gcp-ipi-user-labels-tags-filestore-csi-tp-arm-f28` in openshift-openshift-tests-private-release-4.17__multi-nightly-4.17-upgrade-from-stable-4.17.yaml

File: ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh (hunk 1 above)
Comment (jianli-wei) [reply ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh:64]: I happened noticing the previous way not working, i.e. even I set "ret=1" for a failure within the "while" loop, the function will still return "0". Not quite sure why. Anyway, the new way does work for both failure and success scenarios. Any suggestion or concern? Thanks!

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh:64]: Just be curious why the original way does not work, generally we can `set -x` for debugging to locate the issue. I have no objection on the new change, keep it as your desire.

**Comment by jianlinliu** [comment]
@jianli-wei ci/prow/generated-config job failed.

**Comment by jianli-wei** [comment]
@jianlinliu please help review, thanks!

**Comment by jianlinliu** [comment]
cc @liangxia to review.

File: ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh (hunk 1 above)
Comment (jianli-wei) [reply ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh:64]: Got it, let me try some time later. Thanks!

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.16-installer-rehearse-gcp

File: ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh [hunk 4]
```diff
@@ -29,46 +29,122 @@ then
...
+
+  if [ ${curr_x} -ge ${min_x} ] && [ ${curr_y} -ge ${min_y} ]; then
+    echo "[DEBUG] version_check result: ${ocp_version} >= ${minimum_version}"
+    ret=0
+  else
+    echo "[DEBUG] version_check result: ${ocp_version} < ${minimum_version}"
+    ret=1
+  fi
+
+  rm pull-secret
```
Comment (liangxia) [inline ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh:85]: If any of lines between 70 to 83 failed, the `rm` will be skipped.  Perhaps move the `rm` to line after 68 ?

Comment (jianli-wei) [reply ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh:85]: @liangxia The lines between 70 and 83 don't have much things, and I don't think any will fail.

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.16-installer-rehearse-gcp

**Comment by jianli-wei** [comment]
FYI the 4.16 post-installation check failure was due to https://issues.redhat.com/browse/OCPBUGS-33947.

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-gcp

File: ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh [hunk 5]
```diff
@@ -29,46 +29,123 @@ then
...
+  ocp_version=$(oc adm release info --registry-config pull-secret ${TESTING_RELEASE_IMAGE} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
+
+  echo "[DEBUG] minimum OCP version: '${minimum_version}'"
+  echo "[DEBUG] current OCP version: '${ocp_version}'"
+  curr_x=$(echo "${ocp_version}" | cut -d. -f1)
+  curr_y=$(echo "${ocp_version}" | cut -d. -f2)
+  min_x=$(echo "${minimum_version}" | cut -d. -f1)
+  min_y=$(echo "${minimum_version}" | cut -d. -f2)
+
+  if [ ${curr_x} -gt ${min_x} ] || ([${curr_x} -eq ${min_x} ] && [ ${curr_y} -ge ${min_y} ]); then
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh:78]: I think we even do not need to care about `curr_x`, just comparing `curr_y` and `min_y` should be enough.

Comment (jianli-wei) [reply ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh:78]: But what if comparing "4.17" with "5.1" for example? Thanks!

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-gcp

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-gcp
--


</markdown-accessiblity-table><br class="Apple-interchange-newline">

File: ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh (hunk 5 above)
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh]: Do we have 5.X ocp version ?

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-gcp

File: ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh (hunk 5 above)
Comment (jianli-wei) [reply ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh]: No for now, but I think the function should support comparing 4.m with 5.n. WDYT? Thanks!

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh]: It is too far when we get OCP 5.x,  if we have 5.x, that sounds a different product, I guess most of steps can not be used created for 4.x.

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Comment by jianlinliu** [comment]
@liangxia can you help approve this PR.

File: ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh [hunk 6]
```diff
@@ -9,6 +9,29 @@ if [ ! -f "${SHARED_DIR}/user_tags_sa.json" ]; then
...
 fi
 
+# release-controller always expose RELEASE_IMAGE_LATEST when job configuraiton defines release:latest image
+echo "RELEASE_IMAGE_LATEST: ${RELEASE_IMAGE_LATEST:-}"
+# seem like release-controller does not expose RELEASE_IMAGE_INITIAL, even job configuraiton defines 
+# release:initial image, once that, use 'oc get istag release:inital' to workaround it.
+echo "RELEASE_IMAGE_INITIAL: ${RELEASE_IMAGE_INITIAL:-}"
+if [[ -n ${RELEASE_IMAGE_INITIAL:-} ]]; then
+    tmp_release_image_initial=${RELEASE_IMAGE_INITIAL}
+    echo "Getting inital release image from RELEASE_IMAGE_INITIAL..."
```
Comment (liangxia) [inline ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh:19]: Some of the lines in this file are using 2 white spaces as indention, while the others are using 4 white spaces. Shall we consistently using the same style (at least in one file) ?

**Comment by jianli-wei** [comment]
@jianlinliu could you help adding lgtm again, thanks!

**Pull request merged**